AK/String.cpp

```diff
@@ -125,7 +125,7 @@ String String::isolated_copy() const
 String String::substring(size_t start, size_t length) const
```

Comment:
The commit message refers to `String::split()` but the function changed is `String::substring()`.
Also it would be nice to add a test to `AK/Tests/TestString.cpp` to document this change.

Reply:
I've updated the commit message, and added additional tests for `String::split()` and `String::Substring()`